ledfx/devices/hue.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import socket
+import time
+from typing import Dict, Optional, Tuple
+
+import requests
+import voluptuous as vol
+from mbedtls import exceptions, tls
+
+from ledfx.devices import NetworkedDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HueDevice(NetworkedDevice):
+    """
+    Philips Hue device support (Entertainment Mode UDP streaming)
+    """
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Required(
+                "ip_address",
+                description="Hostname or IP address of the Hue bridge",
+            ): str,
+            vol.Optional("udp_port", description="port", default=2100): int,
+            vol.Required(
+                "user_name",
+                description="User name",
+            ): str,
+            vol.Required(
+                "client_key",
+                description="Client key",
+            ): str,
+            vol.Required(
+                "group_name",
+                description="Entertainment zone group name",
+            ): str,
+        }
+    )
+
+    status: Dict[int, Tuple[int, int, int]]
+    _sock: Optional[socket.socket] = None
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+
+        self._dtls_client_context = tls.ClientContext(
+            tls.DTLSConfiguration(
+                pre_shared_key=(
+                    self._config["user_name"],
+                    bytes.fromhex(self._config["client_key"]),
+                ),
+                ciphers=["TLS-PSK-WITH-AES-128-GCM-SHA256"],
+            )
+        )
+        self.status = {}
+
+    def _hue_get(self, api_endpoint, data=None):
```

Comment:
You can probably remove both hue get and hue put and replace with a single function that you pass the request type to, for example:
```py
def _hue_request(self, method, api_endpoint, data=None):
    return getattr(requests, method)(
        f"http://{self._config['ip_address']}/api/{self._config['user_name']}/{api_endpoint}",
        json=data,
    ).json()
        ```